chrome/content/zotero/zoteroPane.js

```diff
@@ -2786,6 +2786,16 @@ var ZoteroPane = new function()
 				return;
 			}
 			
+			//Special treatment for zotero links
+			if (uri.match(/^zotero\:\/\/(open-pdf|select)/)) {
+				let ioservice = Components.classes["@mozilla.org/network/io-service;1"]
+					  .getService(Components.interfaces.nsIIOService);
+				let zoteroLink = ioservice.newURI(uri, null, null);
+				openZoteroLink = Components.classes["@mozilla.org/network/protocol;1?name=zotero"]
+						.getService(Components.interfaces.nsIProtocolHandler).newChannel(zoteroLink);
+				return;
```

Comment:
What does creating an nsIChannel without calling `.open()` do here? This looks like a no-op to me, but I may be missing something.

Also, couldn't all of this `if` block be just
```
Components.classes["@mozilla.org/network/io-service;1"]
  .getService(Components.interfaces.nsIIOService)
  .newChannel(uri, null, null)
  .open()
```
As per https://developer.mozilla.org/en-US/docs/Mozilla/Tech/XPCOM/Reference/Interface/nsIIOService#newChannel()